tests/test_utils.py

```diff
@@ -298,11 +306,536 @@ def check_multistream_classic(fnames):
+             'OUT_LWNET', 'OUT_SENSIBLE', 'OUT_LATENT', 'OUT_GRND_FLUX']
+
+    # read in data with -9999.0000 as NaNs
+    obs_dir = os.path.join(science_test_data_dir, 'datasets',
+                           'ec_flux_towers', 'obs')
+    ecflux_df = pd.read_csv(os.path.join(obs_dir, subdir, filename),
+                            skiprows=0,
+                            delim_whitespace=True,
+                            header=None,
+                            names=names,
+                            na_values=-9999.0000)
+
+    # add datetime index
+    time_cols = ['YEAR', 'MONTH', 'DAY']
+    ecflux_df.index = pd.to_datetime(ecflux_df[time_cols])
+
+    if 'HOUR' in ecflux_df:
+        ecflux_df.index += pd.Series([pd.Timedelta(s, unit='s') for s in
+                                     ecflux_df['HOUR'] * 3600],
+                                     index=ecflux_df.index)
+        time_cols.append('HOUR')
+
+    # remove year, day columns of DataFrame
+    ecflux_df.drop(time_cols, inplace=True, axis=1)
+
+    '''if ('HOUR' not in ecflux_df) and ('SEC' in ecflux_df):
+        # add hour column for groupby
+        ecflux_df['HOUR'] = (ecflux_df['SEC'] * (1/3600)
+                                        ).astype(int)'''
+
+    return ecflux_df
+
+
+def plot_fluxnet_comparison(driver, science_test_data_dir,
+                            compare_data_dict,
+                            result_dir, plot_dir,
+                            plots_to_make):
+
+    ''' makes Ameriflux figures
+    '''
+
+    # loop over Ameriflux sites
+    obs_dir = os.path.join(science_test_data_dir,
+                           'datasets',
+                           'ec_flux_towers',
+                           'obs')
+
+    for subdir in os.listdir(obs_dir):
+
+        if check_site_files(obs_dir, subdir):
+            # get CSV file from site directory to get lat/lng for site
+            lat, lng = get_fluxnet_lat_lon(obs_dir, subdir)
+
+            # loop over data to compare
+            data = {}
+            for key, items in compare_data_dict.items():
+
+                if key == "ecflux":
+                    try:
+                        # load Ameriflux data
+                        data[key] = read_fluxnet_obs(subdir,
+                                                     science_test_data_dir,
+                                                     items)
+                    except OSError:
+                        warnings.warn("this site does not have data")
+
+                elif key == "VIC.4.2.d":
+                    try:
+                        # load VIC 4.2 simulations
+                        data[key] = read_vic_42_output(lat, lng,
+                                                       science_test_data_dir,
+                                                       items)
+
+                    except OSError:
+                        warnings.warn("this site has a " +
+                                      "lat/lng precision issue")
```

Comment:
The plus sign (`+`) is not needed here.

This would be better syntax though:

```Python
                        warnings.warn(
                            "this site has a lat/lng precision issue")
```